osu.Game/Screens/Edit/Timing/TapTimingControl.cs

```diff
@@ -0,0 +1,104 @@
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Overlays;
+
+namespace osu.Game.Screens.Edit.Timing
+{
+    public class TapTimingControl : CompositeDrawable
+    {
+        [Resolved]
+        private EditorClock editorClock { get; set; }
+
+        [Resolved]
+        private Bindable<ControlPointGroup> selectedGroup { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider, OsuColour colours)
+        {
+            Height = 200;
+            RelativeSizeAxes = Axes.X;
+
+            CornerRadius = LabelledDrawable<Drawable>.CORNER_RADIUS;
+            Masking = true;
+
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    Colour = colourProvider.Background4,
+                    RelativeSizeAxes = Axes.Both,
+                },
+                new GridContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    RowDimensions = new[]
+                    {
+                        new Dimension(),
+                        new Dimension(GridSizeMode.Absolute, 60),
+                    },
+                    Content = new[]
+                    {
+                        new Drawable[]
+                        {
+                            new MetronomeDisplay
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                            }
+                        },
+                        new Drawable[]
+                        {
+                            new Container
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Padding = new MarginPadding(10),
+                                Children = new Drawable[]
+                                {
+                                    new RoundedButton
+                                    {
+                                        Text = "Reset",
+                                        BackgroundColour = colours.Pink,
+                                        RelativeSizeAxes = Axes.X,
+                                        Width = 0.3f,
+                                        Action = reset,
+                                    },
+                                    new RoundedButton
+                                    {
+                                        Anchor = Anchor.TopRight,
+                                        Origin = Anchor.TopRight,
+                                        Text = "Play from start",
+                                        RelativeSizeAxes = Axes.X,
+                                        BackgroundColour = colourProvider.Background1,
+                                        Width = 0.68f,
+                                        Action = tap,
```

Comment:
Bit of a weird name for this action method. Almost thought it was going to implement the tap-to-set-BPM functionality before seeing what button it's attached to.

Reply:
That is what the button will become. I changed it temporarily for getting this PR shipped.